components/ai_chat/core/browser/engine/engine_consumer_conversation_api_unittest.cc

```diff
@@ -32,15 +34,48 @@
 #include "services/network/public/cpp/shared_url_loader_factory.h"
 #include "testing/gmock/include/gmock/gmock.h"
 #include "testing/gtest/include/gtest/gtest.h"
+#include "url/origin.h"
 
 using ::testing::_;
 
 namespace ai_chat {
 
 namespace {
+
 constexpr int kTestingMaxAssociatedContentLength = 100;
+constexpr size_t kChunkSize = 75;
+
+std::pair<std::vector<Tab>, std::vector<std::string>>
+GetMockTabsAndExpectedTabsJsonString(size_t num_tabs) {
+  size_t num_chunks = (num_tabs + kChunkSize - 1) / kChunkSize;
+  std::vector<Tab> tabs;
+  std::vector<std::string> tabs_json_strings;
+  for (size_t i = 0; i < num_chunks; i++) {
+    std::string tabs_json_string = "[";
+    size_t start_suffix = i * kChunkSize;
+    for (size_t j = start_suffix;
+         j < std::min(kChunkSize + start_suffix, num_tabs); j++) {
+      std::string id = base::StrCat({"id", base::NumberToString(j)});
+      std::string title = base::StrCat({"title", base::NumberToString(j)});
+      std::string url = base::StrCat(
+          {"https://www.example", base::NumberToString(j), ".com"});
+      tabs.push_back({id, title, url::Origin::Create(GURL(url))});
+      base::StrAppend(&tabs_json_string,
+                      {R"({\"id\":\")", id, R"(\",\"title\":\")", title,
+                       R"(\",\"url\":\")", url, R"(\"},)"});
```

Comment:
consider using `base::Value` and converting it into json.

Reply:
I'm in favor of my original approach which set up these expected values via string operations directly in tests, the functions we're testing is already using base::Value and converting it into json.